src/core/thread/mle.cpp

```diff
@@ -1724,7 +1726,11 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     // Active Timestamp
     if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
     {
+        const MeshCoP::Dataset local = mNetif.GetActiveDataset().GetLocal();
+        const MeshCoP::Timestamp *localTimestamp = local.GetTimestamp();
+
         VerifyOrExit(activeTimestamp.IsValid(), error = kThreadError_Parse);
+        VerifyOrExit(update || (localTimestamp && (activeTimestamp.Compare(*localTimestamp) == 0)), ;);
```

Comment:
I'm not sure why you need to compare against the local dataset timestamp.  The original intent was to always call `mNetf.GetActiveDataset().Set()` and have that method properly handle everything.  In particular:

1. Only update the local dataset if the received active timestamp is greater.
2. Only update the network dataset if first attaching to the partition or the received active timestamp is greater.